pkg/backend/filestate/store.go

```diff
@@ -232,6 +232,22 @@ func (p *projectReferenceStore) ListProjects(ctx context.Context) ([]tokens.Name
 	return projects, nil
 }
 
+func (p *projectReferenceStore) ProjectExists(ctx context.Context, projectName string) (bool, error) {
+	if projectName == "" {
+		return false, fmt.Errorf("empty project name")
+	}
+
+	path := filepath.Join(StacksDir, projectName)
```

Comment:
I think this might want to be `path.Join` to use `/`, not OS-specific file separators.
(listBucket will probably normalize anyway, but it feels like we can do it right here anyway.)

Reply:
No, I was thinking [path.Join](https://pkg.go.dev/path#Join).

```go
import "path"
```

```suggestion
	p := path.Join(StacksDir, projectName)
```

Be careful with the `path` package, though. If you have a variable named `path` in scope when you're trying to use it, the code won't compile.

Reply:
OK. Agree with the idea.

Btw I found this thread to helpful related to difference use cases between path.Join and filepath.Join

Reply:
Updated https://github.com/pulumi/pulumi/pull/12798/commits/48756f68b1ec77b93f1ada9bc9dba98a5d1ad806